gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageOptions.java

```diff
@@ -34,12 +35,47 @@
   private static final String DEFAULT_PATH_DELIMITER = "/";
 
   private final String pathDelimiter;
+  private final TimeSource timeSource;
   private transient StorageRpc storageRpc;
 
+  /**
+   * A class providing access to the current time in milliseconds.
+   *
+   * Implementations should implement {@code Serializable} wherever possible and must document
+   * whether or not they do support serialization.
+   */
+  public static abstract class TimeSource {
```

Comment:
Though interfaces are *typically* not Serialize I am not sure that convention applies to abstract classes.